test/unit/cmap/commands.test.js

```diff
@@ -0,0 +1,92 @@
+const { expect } = require('chai');
+const { Response } = require('../../../src/cmap/commands');
+
+describe('commands', function () {
+  describe('Response', function () {
+    describe('#parse', function () {
+      context('when the message body is invalid', function () {
```

Comment:
Let's expand the title here, what about the body is invalid? beacuse it's empty correct? We should add a case with invalid data as well, which I think is only the numReturned field (See comment with code).

Reply:
I've broke this down into 2 subcontexts now that say 1) the body is empty and 2) nuReturned is invalid.